### PR TITLE
Updated `cvars` library

### PR DESCRIPTION
**Pull request by PrikolMen**
I updated the outdated `cvars` library code, corrected the comments, replaced depricated functions with actual ones, and improved it by removing unnecessary checks.

I also formatted the entire code according to [CONTRIBUTING.md](https://github.com/Facepunch/garrysmod/blob/master/CONTRIBUTING.md)
All tabs were replaced by 4 spaces, function argument names were fixed, and a few small changes were made.

### CODE REVIEW

**Comment by Kefta** [comment]
I do not see this as an improvement. In fact it introduces regressions, such as as no longer checking that the identifier to cvars.Add/RemoveChangeCallback is a string and cvars.Bool can now return nil.

**Comment by robotboy655** [comment]
[CONTRIBUTING.md](https://github.com/Facepunch/garrysmod/blob/master/CONTRIBUTING.md) specifically states to use `Tab` characters, not spaces.

**Comment by PrikolMen** [comment]
what do you mean, then just write `4 tabs` instead of `tab = 4 spaces`

image

**Comment by PrikolMen** [comment]
> I do not see this as an improvement. In fact it introduces regressions, such as as no longer checking that the identifier to cvars.Add/RemoveChangeCallback is a string and cvars.Bool can now return nil.

Have you used cvars at all? String checking is not needed there at all because it gives nothing and stupidly limits developers and wastes CPU time on checking, there are no operations with strings. And return `nil` is much easier to check than `false` plus it's much more universal.

**Comment by PrikolMen** [comment]
> [CONTRIBUTING.md](https://github.com/Facepunch/garrysmod/blob/master/CONTRIBUTING.md) specifically states to use `Tab` characters, not spaces.

fixed

**Comment by Kefta** [comment]
These are ancient functions with defined behaviours that addons rely on - you can't just change them because you think it's better or more efficient.

**Comment by PrikolMen** [comment]
> These are ancient functions with defined behaviours that addons rely on - you can't just change them because you think it's better or more efficient.

For example for me the behavior of `cvars.Bool` is not at all obvious and not predictable, and I have not seen any addon that checks `cvars.Bool( "cvar" ) == false `, I say more if it causes problems it is extremely easy to correct the developers, so that your statement is silly to say the least.

Personally, I'll tell you an amazing fact, you can't update the game without breaking something, you either don't update it and it dies, or you update it and the developers adapt, you shouldn't look at the "old addons" you should look at how simple, clear and productive it is, unless you want to turn the game into a graveyard of good ideas.

File: garrysmod/lua/includes/modules/cvars.lua [hunk 1]
```diff
@@ -18,111 +11,124 @@ module( "cvars" )
...
+	Desc: Returns a table of the given ConVars callbacks
+-----------------------------------------------------------]]
+function GetConVarCallbacks( name, createIfNotFound )
 
-	if ( CreateIfNotFound && !Tab ) then
-		Tab = {}
-		ConVars[ name ] = Tab
+	local tab = ConVars[ name ]
+	if ( createIfNotFound and !tab ) then
+		tab = {}; ConVars[ name ] = tab
```
Comment (robotboy655) [inline garrysmod/lua/includes/modules/cvars.lua:22]: What's the point of putting these on the same line?

File: garrysmod/lua/includes/modules/cvars.lua [hunk 2]
```diff
@@ -18,111 +11,124 @@ module( "cvars" )
...
+	local tab = GetConVarCallbacks( name )
+	if ( !tab ) then return end
 
-	for k, v in pairs( Callbacks ) do
-
-		if ( istable( v ) ) then
-			v[ 1 ]( name, oldvalue, newvalue )
+	for i = 1, #tab do
+		local callback = tab[ i ]
+		if ( type( callback ) == "table" ) then
```
Comment (robotboy655) [inline garrysmod/lua/includes/modules/cvars.lua:40]: What's the point of changing `istable( v )` calls  to `type( callback ) == "table"`?

File: garrysmod/lua/includes/modules/cvars.lua [hunk 3]
```diff
@@ -18,111 +11,124 @@ module( "cvars" )
...
-	Name: OnConvarChanged
-	Desc: Called by the engine
+	Name: AddChangeCallback
+	Desc: Adds a callback to be called when convar changes
 -----------------------------------------------------------]]
-function AddChangeCallback( name, func, sIdentifier )
-
-	if ( sIdentifier ) then
-		assert( isstring( sIdentifier ), format( "bad argument #%i (string expected, got %s)", 3, type( sIdentifier ) ) )
-	end
```
Comment (robotboy655) [inline garrysmod/lua/includes/modules/cvars.lua:62]: What is the point of removing these? This is a regression that will cause more confusion for the people incorrectly using these functions during development.

**Review by robotboy655 — CHANGES_REQUESTED** [review]
I am seeing a bunch of unnecessary changes here,

File: garrysmod/lua/includes/modules/cvars.lua (hunk 1 above)
Comment (PrikolMen) [reply garrysmod/lua/includes/modules/cvars.lua:22]: just looks shorter and better, i can split it into two lines if you think it's unacceptable

File: garrysmod/lua/includes/modules/cvars.lua (hunk 2 above)
Comment (PrikolMen) [reply garrysmod/lua/includes/modules/cvars.lua:40]: purpose is to make the code shorter, why do we need to localize `istable` if we already have `type`, why do we need both functions?

File: garrysmod/lua/includes/modules/cvars.lua (hunk 3 above)
Comment (PrikolMen) [reply garrysmod/lua/includes/modules/cvars.lua:62]: what is the point of this restriction, what confusion are we talking about if it doesn't affect anything, in fact the identifier can be really anything, there are no operations with string

Comment (PrikolMen) [reply garrysmod/lua/includes/modules/cvars.lua:62]: This value can be `nil` altogether isn't that confusing? Why a value that should be a string can be `nil`, how we can find and remove this value?

File: garrysmod/lua/includes/modules/cvars.lua (hunk 1 above)
Comment (PrikolMen) [reply garrysmod/lua/includes/modules/cvars.lua]: Corrected

File: garrysmod/lua/includes/modules/cvars.lua (hunk 2 above)
Comment (PrikolMen) [reply garrysmod/lua/includes/modules/cvars.lua]: Okay

File: garrysmod/lua/includes/modules/cvars.lua (hunk 3 above)
Comment (PrikolMen) [reply garrysmod/lua/includes/modules/cvars.lua:62]: Returned

**Review by robotboy655 — APPROVED** [review]

**Pull request merged**